src/modules/events-app.module/fields.json

```diff
@@ -1 +1,58 @@
-[]
+[
+  {
+    "label": "Background color",
+    "name": "background_color",
+    "type": "color",
+    "required": false,
+    "locked": false,
+    "default": {}
```

Comment:
My only concern with this change is it's relying on having initially "unset" or `null` defaults for color fields-- the user isn't able to revert back to this initial "inherit" behavior without rolling back versions of the page. 

Said another way, if I make changes to the color fields in-page, but want to revert back to inheriting from whatever theme CSS originally was being applied, I'm sort of out of luck. Reverting to the "default" on each color field sets `#FFFFFF`, which isn't desirable
![Edit_Page___Events](https://user-images.githubusercontent.com/16004483/80031876-526dd500-84b8-11ea-9abd-765cf21d83e6.png)
because the color fields have no developer-defined defaults.

I think this problem isn't unique to this app and I'd be fine with solving for this in a future PR, but without the ability to explicitly set "no color" or "initial/inherit" in a color field ( in either a theme or a module ) you're left with an initial state that isn't able to be reliably recreated by the user. Themes solve this with inherited field values, but for modules that live outside a theme intended to be used in multiple themes, it isn't an option.

We've gotten around this in the past by using a separate boolean / checkbox module field to determine whether the module should use "module-specific" styling or try to inherit from existing CSS. While this works, it tends to be cumbersome and adds a lot of clicks. 

I'm curious what other folks have thoughts about this, too.

Reply:
Hi @levinkeo, I've given some thought to this, and I'm not quite sure what the best solution would be.

Originally, I did have the macro revert back to 'inherit' if there were no color input, but this was causing bugs with the Calendar Highlight Color. Now that we're leaning toward deleting the calendar altogether, perhaps I ought to refactor the conditional to include 'inherit' again. But then I wonder whether this may cause more unexpected behavior and whether it's just a temporary fix. 

I could also set default colors for each of the module fields, but as I wrote above to @gcorne , I'm wondering if this might prove an unpleasant surprise to customers if they import this module and realize it contains its own styling apart from their theme. 

I'm personally leaning toward including a boolean or checkbox, as you suggested above, that would set the color fields to inherit from the theme. I know it's more clicks for the customer, but I think it would also avoid confusion. I can write up a separate issue for this, if that sounds all right to you :)

Reply:
Separate issue sounds good, since I think this will inform non-themed style-able components beyond just this app-- even using the original macro solution utilizing `inherit` poses the same problem in that it relies on checking if a color has been set (since the user can't unset a color / a color doesn't have a `null` option that's settable in the module field UI the content creator interacts with).